mlir/include/mlir/Dialect/Rock/utility/fusionUtils.h

```diff
@@ -0,0 +1,22 @@
+//===- fusionUtils.h - Rock utility for fusion -----------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------===//
+#ifndef ROCK_UTILITY_FISION_H
+#define ROCK_UTILITY_FISION_H
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir {
+namespace rock {
+LogicalResult testFusibility(ModuleOp mod);
+LogicalResult testFusibility(func::FuncOp func);
```

Comment:
I think this might be missing `testFusability(Operation* memoryOp, linalg::GenericOp laGeneric);`

Also, can we rename this guy `testFusionLegality();`?

Reply:
Yes, we can. I will rename it

Reply:
I renamed `testFusability` to `testFusionLegality`. 

I would suggest to implement `testFusionLegality(Operation* memoryOp, linalg::GenericOp laGeneric)` in a dedicated PR.

Reply:
move to a dedicated ticket `testFusionLegality(Operation* memoryOp, linalg::GenericOp laGeneric)`

Reply:
The corresponding ticket has been created. See, [here](https://github.com/ROCm/rocMLIR-internal/issues/1461)